oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/async/ElasticResultRowAsyncIterator.java

```diff
@@ -138,13 +140,38 @@ public void on(Hit<ObjectNode> searchHit) {
             }
             try {
                 queue.put(new FulltextResultRow(path, searchHit.score() != null ? searchHit.score() : 0.0,
-                        null, elasticFacetProvider, null));
+                        readExcerptsFromResponse(searchHit), elasticFacetProvider, null));
             } catch (InterruptedException e) {
                 throw new IllegalStateException("Error producing results into the iterator queue", e);
             }
         }
     }
 
+    /**
+     * Reads excerpts from elasticsearch response.
+     * rep:excerpt and rep:excerpt(.) keys are used for :fulltext
+     * rep:excerpt(PROPERTY) for other fields.
+     * Note: properties to get excerpt from must be included in the _source, which means ingested,
+     * not necessarily Elasticsearch indexed, neither included in the mapping properties.
+     *
+     * @param searchHit
+     * @return a map with the excerpts
+     * @throws IOException
+     */
+    private Map<String, String> readExcerptsFromResponse(Hit<ObjectNode> searchHit) {
+        Map<String, String> excerpts = Maps.newHashMap();
```

Comment:
Guava does not bring any value here. We are trying to minimize its use. Please remove it
```suggestion
        Map<String, String> excerpts = new HashMap<>();
```